compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1457,21 +1457,38 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
         }
       }
 
-      val methodName = abstractMethod.javaSimpleName
-      val applyN = {
-        val mt = asmMethodType(abstractMethod)
-        mt.toASMType
+      val methodName = samMethod.javaSimpleName
+      val samMethodType = asmMethodType(samMethod).toASMType
+      val needsGenericBridge = samMethodType != instantiatedMethodType
```

Comment:
I suggest adding the same comment than in Scala 2 for context:
```suggestion
      // scala/bug#10334: make sure that a lambda object for `T => U` has a method `apply(T)U`, not only the `(Object)Object`
      // version. Using the lambda a structural type `{def apply(t: T): U}` causes a reflective lookup for this method.
      val needsGenericBridge = samMethodType != instantiatedMethodType
```